samples/extensions/full_screen_exclusive/full_screen_exclusive.cpp

```diff
@@ -0,0 +1,1201 @@
+#if defined(VK_USE_PLATFORM_WIN32_KHR)
+	if (input_event.get_source() == vkb::EventSource::Keyboard)
+	{
+		const auto &key_button = reinterpret_cast<const vkb::KeyInputEvent &>(input_event);
+
+		if (key_button.get_action() == vkb::KeyAction::Down)
+		{
+			bool isRecreate = false;
+
+			switch (key_button.get_code())
+			{
+				case vkb::KeyCode::F1:        // FullscreenExclusiveEXT = Disallowed
+					if (full_screen_status != SwapchainMode::Windowed)
+					{
+						full_screen_status                                         = SwapchainMode::Windowed;
+						application_window_status                                  = ApplicationWindowMode::Windowed;
+						surface_full_screen_exclusive_info_EXT.fullScreenExclusive = VK_FULL_SCREEN_EXCLUSIVE_DISALLOWED_EXT;
+						isRecreate                                                 = true;
+						isFullScreenExclusive                                      = false;
+						LOGI("Windowed Mode Detected!")
+					}
+					break;
+				case vkb::KeyCode::F2:        // FullscreenExclusiveEXT = Allowed
+					if (full_screen_status != SwapchainMode::BorderlessFullscreen)
+					{
+						full_screen_status                                         = SwapchainMode::BorderlessFullscreen;
+						application_window_status                                  = ApplicationWindowMode::Fullscreen;
+						surface_full_screen_exclusive_info_EXT.fullScreenExclusive = VK_FULL_SCREEN_EXCLUSIVE_ALLOWED_EXT;
+						isRecreate                                                 = true;
+						isFullScreenExclusive                                      = false;
+						LOGI("Borderless Fullscreen Mode Detected!")
+					}
+					break;
+				case vkb::KeyCode::F3:        // FullscreenExclusiveEXT = Application Controlled
+					if (full_screen_status != SwapchainMode::ExclusiveFullscreen)
+					{
+						full_screen_status                                         = SwapchainMode::ExclusiveFullscreen;
+						application_window_status                                  = ApplicationWindowMode::Fullscreen;
+						surface_full_screen_exclusive_info_EXT.fullScreenExclusive = VK_FULL_SCREEN_EXCLUSIVE_APPLICATION_CONTROLLED_EXT;
+						isRecreate                                                 = true;
+						isFullScreenExclusive                                      = true;
+						LOGI("Exclusive Fullscreen Mode Detected!")
+					}
+					break;
+				default:        // FullscreenExclusiveEXT = Default
+					isRecreate = false;
+					break;
+			}
+
+			// now if to recreate the swapchain and everything related:
+			if (isRecreate)
+			{
+				FullScreenExclusive::recreate();
+			}
+		}
+	}
+#endif
+}
+
+void FullScreenExclusive::update_application_window()
+{
+#if defined(VK_USE_PLATFORM_WIN32_KHR)
+	if (application_window_status == ApplicationWindowMode::Fullscreen && isWindowed)        // check if it is already in fullscreen, if is, then do nothing
+	{
+		isWindowed = false;
+
+		GetWindowPlacement(HWND_applicationWindow, &wpc);
```

Comment:
Can you get the `HWND_style` and `HWND_style_previous` just get in `initialize_windows`?